site/_js/web-components/tag-pill-list.js

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Given a key-value array, renders
+ * a list of tag-pills
+ */
+import {BaseElement} from './base-element';
+import {html} from 'lit-element';
+import {unsafeSVG} from 'lit-html/directives/unsafe-svg';
+import closeIcon from '../../_includes/icons/close.svg';
+
+export class TagPillList extends BaseElement {
+  constructor() {
+    super();
+
+    this.items = [];
+  }
+
+  static get properties() {
+    return {
+      items: {type: Array, reflect: true},
+    };
+  }
+
+  _handleClick(item) {
+    const event = new CustomEvent('remove-pill', {
+      detail: item,
+    });
+
+    this.dispatchEvent(event);
```

Comment:
I think this should just remove the pill, rather than fire an event that has to be handled outside of the component. After all, 1. removing the pill influences the internal state of the component at hand. 2. Pill-removing logic would need to be reimplemented on each page that wants to use the pill-list.

Reply:
Granted re #2 but isn't this standard practice with one-way binding? If we mutate the prop internally parent components wouldn't be aware of the change...

Reply:
Are there parent components usually? This is not a react app.